xrmocap/ops/bottom_up_association/fourdag_associator.py

```diff
@@ -0,0 +1,177 @@
+        else:
+            self.point_selector = point_selector
+        if isinstance(fourd_matching, dict):
+            fourd_matching['logger'] = self.logger
+            self.fourd_matching = build_matching(fourd_matching)
+        else:
+            self.fourd_matching = fourd_matching
+        if isinstance(identity_tracking, dict):
+            identity_tracking['logger'] = self.logger
+            self.identity_tracking = build_identity_tracking(identity_tracking)
+        else:
+            self.identity_tracking = identity_tracking
+
+    def set_cameras(
+        self, cameras: List[Union[FisheyeCameraParameter,
+                                  PinholeCameraParameter]]
+    ) -> None:
+        
+        self.triangulator.set_cameras(cameras)
+        if hasattr(self.point_selector, 'triangulator'):
+            self.point_selector.triangulator.set_cameras(cameras)
+        self.fourd_matching.set_cameras(cameras)
+        self.n_views = len(cameras)
+
+
+    def cal_keypoints2d(self, m_personsMap, kps2d_paf):
+    
+        for i, person_id in enumerate(m_personsMap.copy()):
+            if i < len(self.last_multi_kps3d):
+                continue
+            if  sum(sum(m_personsMap[person_id] >= 0)) >= self.min_asgn_cnt:
+                continue
+            else:
+                m_personsMap.pop(person_id)
+
+        m_skels2d = {}
+        for person_id in m_personsMap:
+            if person_id  < len(self.last_multi_kps3d):
+                identity = person_id
+            elif len(m_skels2d) == 0:
+                identity = 0
+            else:
+                identity = max(m_skels2d) + 1
+            skel2d = np.zeros((3,self.n_views *self.n_kps))
+            for view in range(self.n_views):
+                for joint_id in range(self.n_kps):
+                    index = m_personsMap[person_id][joint_id, view]
+                    if index != -1:
+                        skel2d[:,view*self.n_kps+joint_id] = kps2d_paf[view]['joints'][joint_id][index]
+                    else:
+                        continue
+
+            m_skels2d[identity] = skel2d
+        return m_skels2d
+
+    def associate_frame(
+            self, kps2d_paf:list
+    ) -> Tuple[Keypoints, List[int]]:
+        """Associate and triangulate keypoints2d in one frame.
+
+        Args:
+            
+        Returns:
+
+            keypoints3d (Keypoints):
+                An instance of class keypoints,
+                triangulated from the selected
+                keypoints2d.
+            indentities (List[int]):
+                A list of indentities, whose length.
+        """
+        self.n_kps = len(kps2d_paf[0]['joints'])
+        m_personsMap = self.fourd_matching(kps2d_paf, self.last_multi_kps3d)
+        m_skels2d = self.cal_keypoints2d(m_personsMap, kps2d_paf)
+        multi_kps3d = []
+        identities = []
+        for person_id in m_skels2d:
+            keypoints2d = np.zeros((self.n_views, self.n_kps, 3))
```

Comment:
`keypoints` represents an instance of the Keypoints class, and `kps2d/kps3d` represents location, so it should be changed to `kps2d = np.zeros((self.n_views, self.n_kps, 3))`

Reply:
i will resolve it